lib/browser/auth.ts

```diff
@@ -11,17 +11,37 @@
  * @preferred
  */
 
+import { AwsSigningConfigBase } from '../common/auth';
+
+ 
 /**
  * Standard AWS Credentials
  *
  */
-export interface AWSCredentials {
+export interface AWSCredentials{
     /** Optional region */
     aws_region?: string,
     /** AWS access id */
     aws_access_id: string,
     /** AWS secret access key */
     aws_secret_key: string,
     /** Session token for session credentials */
-    aws_sts_token?: string
+    aws_sts_token?: string,
+    /** external credential provider */
+    aws_provider ?: any
+}
+
+/**
+ * Configuration for use in browser credential
+ *
+ * @category Auth
+ */
+export interface AwsSigningConfig extends AwsSigningConfigBase{
```

Comment:
Why does this need to be exported?  It's not something we'd ever want external users to submit given its restriction to just websocket handshakes